tests/test_transformer/test_nodeFusion.2/test_fusion_transformer.py

```diff
@@ -0,0 +1,327 @@
+import tensorflow as tf
+from utensor_cgen.ir import uTensorGraph, OperationInfo
+from copy import deepcopy
+
+def get_ops_io_info(op_type):
+#please refer to OperatorFactory() in operators.py
+  ops_io_table = dict()
+  ops_io_table["Add"] =                 [[0, 0], [0]]
+  ops_io_table["ArgMax"] =              [[0, 1], [0]]
+  ops_io_table["Dequantize"] =          [[0, 1, 2], [0]]
+  ops_io_table["Max"] =                 [[0, 1], [0]]
+  ops_io_table["QuantizedMaxPool"] =    [[0, 1, 2], [0, 1, 2]]
+  ops_io_table["Min"] =                 [[0, 1], [0]]
+  ops_io_table["QuantizeV2"] =          [[0, 1, 2], [0, 1, 2]]
+  ops_io_table["QuantizedMatMul"] =     [[0, 1, 2, 3, 4, 5], [0, 1, 2]]
+  ops_io_table["QuantizedRelu"] =       [[0, 1, 2], [0, 1, 2]]
+  ops_io_table["QuantizedAdd"] =        [[0, 0, 1, 2, 4, 5], [0, 1, 2]]
+  ops_io_table["RequantizationRange"] = [[0, 1, 2], [0, 1]]
+  ops_io_table["Requantize"] =          [[0, 1, 2, 3, 4], [0, 1, 2]]
+  ops_io_table["Reshape"] =             [[0, 1], [0]]
+  ops_io_table["QuantizedReshape"] =    [[0, 1, 2, 3], [0, 1, 2]]
+  ops_io_table["QuantizedConv2D"] =     [[0, 1, 2, 3, 4, 5], [0, 1, 2]]
+  ops_io_table["Const"] =               [None, [0]]
+  ops_io_table["Placeholder"] =         [None, [0]]
+  ops_io_table["Inline"] =              [None, [0]]
+
+  return ops_io_table[op_type]
+
+
+def print_topo_order(graph):
+  for it_node in graph.topo_order:
+    #print(it_node)
+    print(it_node, " : ", graph.ops_info[it_node].op_type, "| ", [t.name for t in graph.ops_info[it_node].input_tensors], ":", [t.name for t in graph.ops_info[it_node].output_tensors])
+  [exp_inputs, exp_outputs] = subgraph_trace_exposed_edges(graph)
+  print("exposed inputs", exp_inputs)
+  print("exposed outputs", exp_outputs)
+
+def compare_topological_orders(graph0, graph1):
+  if(len(graph0.topo_order) != len(graph1.topo_order)):
+    print("======graph0 topo")
+    print_topo_order(graph0)
+    print("======graph1 topo")
+    print_topo_order(graph1)
+    return False
+
+  for node0, node1 in zip(graph0.topo_order, graph1.topo_order):
+    op_type0 = graph0.ops_info[node0].op_type
+    op_type1 = graph1.ops_info[node1].op_type
+    if(op_type0 != op_type1):
+      print("======graph0 topo")
+      print_topo_order(graph0)
+      print("======graph1 topo")
+      print_topo_order(graph1)
+      return False
+  return True
+
+def get_input_nodes(graph, node):
```

Comment:
`get_input_nodes` and `get_output_nodes` can be replaced by `graph.ops_info[node].input_nodes` and `graph.ops_info[node].output_nodes` respectively.